ddpui/core/ai/query_executor.py

```diff
@@ -0,0 +1,484 @@
+"""
+Dynamic Query Execution Engine for AI-Generated Queries
+
+This service safely executes AI-generated SQL queries with comprehensive
+safety controls, monitoring, and result formatting.
+"""
+
+import time
+import logging
+from typing import Dict, List, Any, Optional, Tuple
+from dataclasses import dataclass, field
+from datetime import datetime
+import json
+
+from django.utils import timezone
+from sqlalchemy import text
+from sqlalchemy.exc import SQLAlchemyError, TimeoutError as SQLTimeoutError
+
+from ddpui.core.ai.query_generator import (
+    QueryPlan,
+    QueryValidationResult,
+    NaturalLanguageQueryService,
+)
+from ddpui.core.ai.data_intelligence import DataIntelligenceService
+from ddpui.models.org import Org, OrgWarehouse
+from ddpui.datainsights.warehouse.warehouse_factory import WarehouseFactory
+from ddpui.utils.custom_logger import CustomLogger
+
+logger = CustomLogger("ddpui.core.ai.query_executor")
+
+
+@dataclass
+class QueryExecutionResult:
```

Comment:
where is rate limiting happening ? at the warehouse ?
also where are we storing this information & what is the error being raise if we hit the rate limit ?

do we really need rate limiting at this point ? if not , should we remove the piece of logic since this is a beta version ?